File: demo/read_hdf5.m
% 读取h5数据
data = h5read('test.h5','/IF1509_20150202');

% 字符串要转置再处理一下
data.Symbol = cellstr(data.Symbol');
data.Exchange = cellstr(data.Exchange');

t = struct2table(data);

% 先将int32转int64
t.datetime = int64(t.ActionDay)*1000000 + int64(t.UpdateTime);

% 可以再转存成csv
writetable(t,'t.csv');

File: demo/read_hdf5.py
# -*- coding: utf-8 -*-

################
# 使用pandas来读取
import pandas as pd
df1 = pd.read_hdf(r'test.h5','IF1509_20150202')
df1.dtypes

################
# 使用h5py来读取
import h5py
f = h5py.File(r'test.h5', 'a')

# 打印表名
for name in f:
    print(name)

ds = f['IF1509_20150202'][:]

df2 = pd.DataFrame(ds)
df2.dtypes

################
# 时间处理,可看情况是否加入毫秒
import numpy as np
df2['datetime'] = df2['ActionDay'].astype(np.int64)*1000000+df2['UpdateTime']

